apps/buildog/src/components/user-sign-up-form.tsx

```diff
@@ -27,20 +29,32 @@ export function SignUpForm({ className, ...props }: UserAuthFormProps) {
   const [loading, setLoading] = useState<boolean>(false);
   const [verifyYourEmail, setVerifyYourEmail] = useState<boolean>(false);
   const [error, setError] = useState<String>(null);
+  const { toast } = useToast();
 
   const form = useForm<ProfileFormValues>({
     resolver: zodResolver(profileFormSchema),
     mode: "onSubmit",
   });
+  //extracts the error code
+  const extractErrorCode = (errorMessage: string): string => {
+    const match = errorMessage.match(/\(auth\/[a-zA-Z0-9\-]+\)/);
+    return match ? match[0].replace(/[()]/g, "") : "unknown-error";
+  };
```

Comment:
could you please create a file in lib folder? let's call it `firebase-helper.tsx` then we can move this function in that file because we use this function in login as well.